lib/org/openpsa/invoices/billing/data.php

```diff
@@ -33,6 +33,57 @@ public function _on_creating()
     }
 
     /**
+     * Get the billing data for the customer in object or create a default data set of it.
+     *
+     * @param org_openpsa_invoices_interfaces_customer $object
+     * @return org_openpsa_invoices_billing_data_dba|boolean
+     */
+    public static function get_by_object(org_openpsa_invoices_interfaces_customer $object)
+    {
+        $customer = $object->get_customer();
```

Comment:
Since we have the customer object already at this point, it is wasteful to call `self::get_billing_data` twice in the lines below (since they will load the object again from the DB). So either the `$object->get_customer` call should be removed, or the logic below should be adapted to actually use it (but you have to be careful, an invoice/salesproject might have both `customer` and `customerContact`, but it could be that only one of them has billng data)